content/blog/2022-02-28-using-iterative-terraform-provider-with-docker.md

```diff
@@ -0,0 +1,204 @@
+some help from the [Iterative Terraform Provider](https://registry.terraform.io/providers/iterative/iterative/latest). We've 
+talked about [Terraform](https://www.terraform.io), Iterative Terraform Provider and how you can configure the `iterative_task` 
+resource to provision infrastructure, sync data and run your training script. 
+
+To make it easy to get started, we used a pre-configured [Deep Learning AMI from AWS](https://aws.amazon.com/machine-learning/amis/) 
+to simplify the environment setup. In most cases, however, instead of using a pre-configured Deep Learning AMI, you will 
+ship your environment to a remote machine using [Docker](https://www.docker.com) which has become an essential tool not only 
+in traditional software development, but also in machine learning. If you're not sure why it would be a good idea to use 
+Docker for machine learning development, check out 
+[this article from AWS](https://aws.amazon.com/blogs/opensource/why-use-docker-containers-for-machine-learning-development/) 
+that should answer this question for you. As a side note, using Docker to encapsulate your environment doesn't mean you won't
+ be using AMIs at all. In fact, AMI (Amazon Machine Image) is actually a required component of setting up an EC2, and at the 
+ minimum you will need an Ubuntu distribution AMI (which is a configurable default for the `iterative_task`). However, we will 
+ use Docker for the machine learning specific setup: drivers, GPU-specific libraries for accelerated math, and Python libraries. 
+ This separation will not only give you more control over the versions of the frameworks that you choose to use (or not use), 
+ but also over the cloud provider you provision the resources with. Your Docker image is not cloud provider specific while a 
+ Deep Learning AMI is a part of the AWS ecosystem.  
+
+If you have a DevOps team, they can build a Docker image suited for your needs, so you'll only need to use it in your 
+`iterative_task` script. We'll show you how to do this in the first part of this tutorial.  But if you need (or want) to build 
+your own Docker image, we've got you too! The second part of this tutorial will illustrate some basics required for building 
+and using your own Docker image. Without further ado, let's get into it! 
+
+## Setting up environment for utilizing GPUs with Docker
+If you haven't read the [previous tutorial](2022-01-15-moving-local-experiments-to-the-cloud-with-iterative-terraform-provider.md)
+yet and are not familiar with Terraform, I would encourage you to check it out and learn some basics, i.e. how to let Terraform 
+know that you'll be using the Iterative Provider, how to initialize the working directory with `terraform init`, how to use 
+other essential Terraform commands (`plan`, `apply`, `refresh`, `show`, `destroy`). All these will apply here, we'll only modify 
```

Comment:
did we mention `plan`?

Reply:
Yes, I did mention `plan` in the first tutorial.